tabular/src/autogluon/tabular/configs/config_helper.py

```diff
@@ -0,0 +1,357 @@
+        Set this = True if you are willing to tolerate longer training times in order to maximize predictive accuracy!
+        Automatically sets `num_bag_folds` and `num_stack_levels` arguments based on dataset properties.
+        Note: Setting `num_bag_folds` and `num_stack_levels` arguments will override `auto_stack`.
+        Note: This can increase training time (and inference time) by up to 20x, but can greatly improve predictive performance.
+        """
+        self.config['auto_stack'] = auto_stack
+        return self
+
+    def num_bag_folds(self, num_bag_folds: int) -> ConfigBuilder:
+        """
+        Number of folds used for bagging of models. When `num_bag_folds = k`, training time is roughly increased by a factor of `k` (set = 0 to disable bagging).
+        Disabled by default (0), but we recommend values between 5-10 to maximize predictive performance.
+        Increasing num_bag_folds will result in models with lower bias but that are more prone to overfitting.
+        `num_bag_folds = 1` is an invalid value, and will raise a ValueError.
+        Values > 10 may produce diminishing returns, and can even harm overall results due to overfitting.
+        To further improve predictions, avoid increasing `num_bag_folds` much beyond 10 and instead increase `num_bag_sets`.
+        """
+        assert num_bag_folds >= 0, 'num_bag_folds must be greater or equal than zero'
+        self.config['num_bag_folds'] = num_bag_folds
+        return self
+
+    def num_bag_sets(self, num_bag_sets: int) -> ConfigBuilder:
+        """
+        Number of repeats of kfold bagging to perform (values must be >= 1). Total number of models trained during bagging = `num_bag_folds * num_bag_sets`.
+        Defaults to 1 if `time_limit` is not specified, otherwise 20 (always disabled if `num_bag_folds` is not specified).
+        Values greater than 1 will result in superior predictive performance, especially on smaller problems and with stacking enabled (reduces overall variance).
+        """
+        assert num_bag_sets > 0, 'num_bag_sets must be greater than zero'
+        self.config['num_bag_sets'] = num_bag_sets
+        return self
+
+    def num_stack_levels(self, num_stack_levels: int) -> ConfigBuilder:
+        """
+        Number of stacking levels to use in stack ensemble. Roughly increases model training time by factor of `num_stack_levels+1` (set = 0 to disable stack ensembling).
+        Disabled by default (0), but we recommend values between 1-3 to maximize predictive performance.
+        To prevent overfitting, `num_bag_folds >= 2` must also be set or else a ValueError will be raised.
+        """
+        assert num_stack_levels >= 0, 'num_stack_levels must be greater or equal than zero'
+        self.config['num_stack_levels'] = num_stack_levels
+        return self
+
+    def holdout_frac(self, holdout_frac: float) -> ConfigBuilder:
```

Comment:
General comment: Do we allow `None` (the default). Currently it would error in assert.

Reply:
I use assumption that if you want default, you don't use this call.

Reply:
ok, I suppose this is a 2-way-door for now if we start with not allowing.